upload-s3-artifact/action.yml

```
@@ -11,6 +11,13 @@ inputs:
       The S3 key to use when uploading the artifact to S3.
 
       Default: the filename is the SHA-256 checksum of the file to upload.
+  aws-s3-key-prefix:
+    description: |
+      The prefix for the S3 key to use when uploading the artifact to S3.
```

I would suggest making it clearer that this is an optional input
```suggestion
    default: ""
    description: |
      An optional S3 key prefix to use when uploading the artifact to S3.
      
      A use-case for this input is adding an additional custom prefix to the default S3 key (a SHA-256 checksum of the artifact).
```